templates/maillog.tpl

```diff
@@ -0,0 +1,326 @@
+        </thead>
+        <tbody id="content_logs">             
+        </tbody>
+    </table> 
+    
+    <div id="error_msg" style="margin-bottom: 50px;"></div>  
+    <div id="pages" style="margin-bottom: 50px;"></div>
+</div>
+
+
+
+<script>
+    let CURRENT_PAGE = 1;
+    function httpGET(url){
+        var xmlHttp = new XMLHttpRequest();
+        console.log(url);
+        xmlHttp.open( "GET", url, false ); 
+        xmlHttp.send( null );
+        return xmlHttp.responseText;
+    }
+
+    function setLogs(logs, table){
+        
+        document.getElementById("content_logs").innerHTML="";
+        document.getElementById("error_msg").innerHTML="";
+        try {
+            logs = JSON.parse(logs);
+        } catch (error) {
+            document.getElementById("error_msg").innerHTML=`
+                <div class="alert alert-danger" role="alert">
+                    <h4 class="alert-heading">Error to load data</h4>
+                    <hr>
+                    <p class="mb-0">
+                        `+logs+`
+                    </p>
+                </div>     
+            `;
+            
+            return;
+        }
+       
+        if(logs["data"].length == 0 ){
+            document.getElementById("content_logs").innerHTML="No records found!";
+            return;
+        }
+
+        for (let i=0; i<logs["data"].length; i++){
+            if (table == "auth_logs"){    
+                document.getElementById("content_logs").innerHTML+=`            
+                    <tr>
+                        <td style="width:5%">   `+logs["data"][i]["month"]  +`</td>
+                        <td style="width:5%">   `+logs["data"][i]["day"]    +`</td>
+                        <td style="width:5%">   `+logs["data"][i]["hour"]   +`</td>
+                        <td style="width:20%">  `+logs["data"][i]["domain"] +`</td>
+                        <td style="width:20%">  `+logs["data"][i]["ip"]     +`</td>
+                        <td style="width:20%">  `+logs["data"][i]["email"]  +`</td>
+                        <td style="width:20%">  `+logs["data"][i]["log"]    +`</td>
+                    </tr>         
+                `;
+                continue;
+            }
+            if (table == "dovecot_logs"){    
+                document.getElementById("content_logs").innerHTML+=`
+                    <tr>
+                        <td style="width:5%">`  +logs["data"][i]["month"]   +`</td>
+                        <td style="width:5%">`  +logs["data"][i]["day"]     +`</td>
+                        <td style="width:5%">`  +logs["data"][i]["hour"]    +`</td>
+                        <td style="width:20%">` +logs["data"][i]["domain"]  +`</td>
+                        <td style="width:20%">` +logs["data"][i]["email"]   +`</td>
+                        <td style="width:20%">` +logs["data"][i]["msgid"]   +`</td>
+                        <td style="width:20%">` +logs["data"][i]["log"]     +`</td>
```

Comment:
is this is going to be vulnerable to XSS ?

Reply:
no. the data will be available just for the user.